modules/objdetect/misc/python/test/test_qrcode_detect.py

```diff
@@ -50,3 +50,14 @@ def test_detect_and_decode_multi(self):
         self.assertTrue("STEP BACK" in decoded_data)
         self.assertTrue("QUESTION" in decoded_data)
         self.assertEqual(points.shape, (6, 4, 2))
+
+    def test_decode_non_utf8(self):
+        import sys
+        if sys.version_info[0] < 3:
+            raise unittest.SkipTest('Python 2.x is not supported')
+
+        img = cv.imread(os.path.join(self.extraTestDataPath, 'cv/qrcode/umlaut.png'))
+        self.assertFalse(img is None)
+        detector = cv.QRCodeDetector()
+        decoded_data, _, _ = detector.detectAndDecode(img)
+        self.assertTrue(u"M\u00FCllheimstrasse" in decoded_data)
```

Comment:
Please move this test to C++.

Python tests verify bindings-related stuff: functions, passed input parameters, result types.
Python tests should not check result accuracy (one E2E test above is enough). Accuracy checks should be done in C++ tests.